### PR TITLE
dawood/i18n

### PR DESCRIPTION
**Pull request by dawoodkhan82**
# Description

Please include: 
* relevant motivation
* a summary of the change 
* which issue is fixed. 
* any additional dependencies that are required for this change.

Fixes: #361 

- [ ] I have performed a self-review of my own code
- [ ] My code follows the style guidelines of this project
- [ ] I have commented my code in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Comment by abidlabs** [comment]
Cool!! Can you include some screenshots? Will promote this on Twitter

**Comment by dawoodkhan82** [comment]
@pngwn Not sure why I'm getting formatting errors on the check, but not locally when I run it.

**Comment by pngwn** [comment]
Are you running `pnpm format:check` in the `ui` folder? I am getting the same errors locally. If you run `pnpm format:write` it should fix all of these formatting errors.

I took a look at the PR, and it looks good to me other than the failing CI checks.

**Comment by pngwn** [comment]
I had to pull the latest changes in order to replicate the CI failures, they weren't showing up previously.

**Comment by dawoodkhan82** [comment]
> Are you running `pnpm format:check` in the `ui` folder? I am getting the same errors locally. If you run `pnpm format:write` it should fix all of these formatting errors.
> 
> I took a look at the PR, and it looks good to me other than the failing CI checks.

Got it, so i guess it does additional checks than prettier --check

**Comment by pngwn** [comment]
Ah yes. It applies some plugins so that prettier can understand svelte files. Otherwise prettier will ignore them.

**Comment by abidlabs** [comment]
@dawoodkhan82 can you include some screenshots of before vs. after? Will review the PR but want to make sure I know what to look for

**Comment by dawoodkhan82** [comment]
> @dawoodkhan82 can you include some screenshots of before vs. after? Will review the PR but want to make sure I know what to look for
Screen Shot 2022-02-14 at 4 22 53 PM
Screen Shot 2022-02-14 at 4 22 59 PM
Screen Shot 2022-02-07 at 3 29 32 PM

**Comment by abidlabs** [comment]
Thanks, gotcha. Why are there so many formatting related changes? For example, all of the single quotes with double quotes and the "star" bullets with "dash" bullets?

**Comment by dawoodkhan82** [comment]
> Thanks, gotcha. Why are there so many formatting related changes? For example, all of the single quotes with double quotes and the "star" bullets with "dash" bullets?

Thats from running the code formatter on the whole repo. Reverted those changes.

**Comment by abidlabs** [comment]
Looks nice! 

Let's say we wanted to add another language, say Urdu, do we just create a json file here with the translations? 
https://github.com/gradio-app/gradio/tree/dawood/i18n/ui/packages/app/public/lang 

What would the name of the file be and would it have to referenced anywhere else?

**Comment by dawoodkhan82** [comment]
> Looks nice!
> 
> Let's say we wanted to add another language, say Urdu, do we just create a json file here with the translations? https://github.com/gradio-app/gradio/tree/dawood/i18n/ui/packages/app/public/lang
> 
> What would the name of the file be and would it have to referenced anywhere else?

Exactly, create a new file for the language. The file name should be the language code (can easily be looked up). Only needs to be imported and added to this file `ui/packages/app/src/i18n.js`

File: website/homepage/src/guides_template.html [hunk 1]
```diff
@@ -111,6 +111,23 @@
       </div>
       {% endfor %}
       </div>
+    <p class='mb-2 text-sm text-gray-500'>
```
Comment (aliabd) [inline website/homepage/src/guides_template.html:114]: @abidlabs @dawoodkhan82 why was this added again? Accident? 

Screen Shot 2022-02-17 at 3 55 42 PM

Will make a PR to revert but it got deployed to the website so curious how it happened

Comment (abidlabs) [reply website/homepage/src/guides_template.html:114]: Ah gosh this slipped through the PR review. Seems like we must have been working with an old version of the files and for some reason git overwrote `master`!

**Comment by Zengyf-CVer** [comment]
@abidlabs 
How can I change the language, for example, I want to change to `es.json`? Should I change any parameters?

**Comment by abidlabs** [comment]
Hi @Zengyf-CVer, you should change your browser's language locale and the interface language will update automatically: e.g. for Chrome: https://stackoverflow.com/questions/37221494/how-to-change-the-locale-in-chrome-browser

**Comment by Zengyf-CVer** [comment]
@abidlabs 
I mean how can I set the language of the gradio component by using the JSON file inside it? thanks.
https://github.com/gradio-app/gradio/tree/main/ui/packages/app/src/lang
ksnip_20220510-185952

**Comment by dawoodkhan82** [comment]
@Zengyf-CVer These json files are automatically used by your browser. There is no need to do anything with the Gradio library to set the language. Just change the language setting in your browser. Does that answer your question?
